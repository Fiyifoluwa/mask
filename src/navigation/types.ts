import {StackScreenProps} from '@react-navigation/stack';
import {InventoryItem} from '../types';
import {NavigationProp} from '@react-navigation/native';

export type AppStackParamList = {
  AddItem: undefined;
  EditItem: {
    item: InventoryItem;
  };
  Home: undefined;
};

export type AppStackScreenProps<T extends keyof AppStackParamList> =
  StackScreenProps<AppStackParamList, T>;

export type AppStackNavigationType = NavigationProp<AppStackParamList>;
