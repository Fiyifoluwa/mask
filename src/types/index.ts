export interface InventoryItem {
  id: string;
  name: string;
  totalStock: number;
  price: number;
  description: string;
}

export type InventoryAction =
  | {type: 'SET_ITEMS'; payload: InventoryItem[]}
  | {type: 'ADD_ITEM'; payload: InventoryItem}
  | {type: 'UPDATE_ITEM'; payload: InventoryItem}
  | {type: 'DELETE_ITEM'; payload: string};

export interface InventoryState {
  items: InventoryItem[];
  loading: boolean;
  error: string | null;
}
