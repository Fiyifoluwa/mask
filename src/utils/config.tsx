import React from 'react';
import Box from '../components/Box';
import Text from '../components/Text';
import {ToastConfigParams} from 'react-native-toast-message';

interface ToastProps {
  text: string;
}

export const toastConfig = {
  successToast: ({text1, props}: Partial<ToastConfigParams<ToastProps>>) => (
    <Box px="xxl" py="sml" bg="malachite100" mt="ll" borderRadius="m">
      {text1 && <Text color="white">{text1}</Text>}
      <Text color="malachite800" textAlign="center" variant="toastText">
        {props?.text ?? ''}
      </Text>
    </Box>
  ),
  infoToast: ({text1, props}: Partial<ToastConfigParams<ToastProps>>) => (
    <Box px="xxl" py="sml" bg="ochre100" mt="ll" borderRadius="m">
      {text1 && <Text color="white">{text1}</Text>}
      <Text color="ochre800" textAlign="center" variant="toastText">
        {props?.text ?? ''}
      </Text>
    </Box>
  ),
  dangerToast: ({text1, props}: Partial<ToastConfigParams<ToastProps>>) => (
    <Box px="xxl" py="sml" bg="cardinal50" mt="ll" borderRadius="m">
      {text1 && <Text color="white">{text1}</Text>}
      <Text color="red" textAlign="center" variant="toastText">
        {props?.text ?? ''}
      </Text>
    </Box>
  ),
};
