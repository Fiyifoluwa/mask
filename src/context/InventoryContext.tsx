import React, {
  createContext,
  useContext,
  useReducer,
  useEffect,
  Dispatch,
  ReactNode,
  FC,
} from 'react';
import {StorageService} from '../utils/storage';
import {InventoryAction, InventoryItem, InventoryState} from '../types';

const initialState: InventoryState = {
  items: [],
  loading: false,
  error: null,
};

const InventoryContext = createContext<
  | {
      state: InventoryState;
      dispatch: Dispatch<InventoryAction>;
      loadItems: () => Promise<void>;
      addItem: (item: InventoryItem) => Promise<void>;
      updateItem: (item: InventoryItem) => Promise<void>;
      deleteItem: (id: string) => Promise<void>;
    }
  | undefined
>(undefined);

const inventoryReducer = (
  state: InventoryState,
  action: InventoryAction,
): InventoryState => {
  switch (action.type) {
    case 'SET_ITEMS':
      return {
        ...state,
        items: action.payload,
        loading: false,
      };
    case 'ADD_ITEM':
      return {
        ...state,
        items: [...state.items, action.payload],
      };
    case 'UPDATE_ITEM':
      return {
        ...state,
        items: state.items.map(item =>
          item.id === action.payload.id ? action.payload : item,
        ),
      };
    case 'DELETE_ITEM':
      return {
        ...state,
        items: state.items.filter(item => item.id !== action.payload),
      };
    default:
      return state;
  }
};

export const InventoryProvider: FC<{children: ReactNode}> = ({children}) => {
  const [state, dispatch] = useReducer(inventoryReducer, initialState);

  const loadItems = async () => {
    try {
      const items = await StorageService.getItems();
      dispatch({type: 'SET_ITEMS', payload: items});
    } catch (error) {
      console.error('Error loading items:', error);
    }
  };

  const addItem = async (item: InventoryItem) => {
    try {
      await StorageService.addItem(item);
      dispatch({type: 'ADD_ITEM', payload: item});
    } catch (error) {
      console.error('Error adding item:', error);
    }
  };

  const updateItem = async (item: InventoryItem) => {
    try {
      await StorageService.updateItem(item);
      dispatch({type: 'UPDATE_ITEM', payload: item});
    } catch (error) {
      console.error('Error updating item:', error);
    }
  };

  const deleteItem = async (id: string) => {
    try {
      await StorageService.deleteItem(id);
      dispatch({type: 'DELETE_ITEM', payload: id});
    } catch (error) {
      console.error('Error deleting item:', error);
    }
  };

  useEffect(() => {
    loadItems();
  }, []);

  return (
    <InventoryContext.Provider
      value={{
        state,
        dispatch,
        loadItems,
        addItem,
        updateItem,
        deleteItem,
      }}>
      {children}
    </InventoryContext.Provider>
  );
};

export const useInventory = () => {
  const context = useContext(InventoryContext);
  if (context === undefined) {
    throw new Error('useInventory must be used within an InventoryProvider');
  }
  return context;
};
