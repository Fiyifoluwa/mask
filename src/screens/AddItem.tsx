import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {useInventory} from '../context/InventoryContext';
import {InventoryForm} from '../components/InventoryForm';
import {InventoryItem} from '../types';

export const AddItem = () => {
  const navigation = useNavigation();
  const {
    addItem,
    state: {items},
  } = useInventory();

  const handleSubmit = async (values: Partial<InventoryItem>) => {
    const newItem: InventoryItem = {
      id: Date.now().toString(),
      name: values.name!,
      totalStock: Number(values.totalStock),
      price: Number(values.price),
      description: values.description!,
    };

    await addItem(newItem);
    navigation.goBack();
  };

  return (
    <InventoryForm
      initialValues={{
        name: '',
        totalStock: 0,
        price: 0,
        description: '',
      }}
      onSubmit={handleSubmit}
      submitButtonText="Add Item"
      existingItems={items}
    />
  );
};
