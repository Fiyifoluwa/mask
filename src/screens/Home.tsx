import React from 'react';
import {View, Text, FlatList, TouchableOpacity, StyleSheet} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {useInventory} from '../context/InventoryContext';
import {InventoryItem} from '../types';

export const Home = () => {
  const {
    state: {items},
  } = useInventory();
  const navigation = useNavigation();

  const renderItem = ({item}: {item: InventoryItem}) => (
    <TouchableOpacity
      style={styles.itemContainer}
      onPress={() => navigation.navigate('EditItem', {item})}>
      <Text style={styles.itemName}>{item.name}</Text>
      <Text>Stock: {item.totalStock}</Text>
      <Text>Price: ${item.price.toFixed(2)}</Text>
    </TouchableOpacity>
  );

  return (
    <View style={styles.container}>
      <FlatList
        data={items}
        renderItem={renderItem}
        keyExtractor={item => item.id}
        ListEmptyComponent={
          <Text style={styles.emptyText}>No items in inventory</Text>
        }
      />
      <TouchableOpacity
        style={styles.addButton}
        onPress={() => navigation.navigate('AddItem')}>
        <Text style={styles.addButtonText}>Add Item</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    backgroundColor: '#fff',
  },
  itemContainer: {
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
  },
  itemName: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  emptyText: {
    textAlign: 'center',
    marginTop: 20,
    fontSize: 16,
    color: '#666',
  },
  addButton: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    backgroundColor: '#007AFF',
    padding: 16,
    borderRadius: 30,
    elevation: 4,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.25,
    shadowRadius: 4,
  },
  addButtonText: {
    color: '#fff',
    fontWeight: 'bold',
  },
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 4,
    padding: 8,
    marginBottom: 16,
  },
  multiline: {
    height: 100,
    textAlignVertical: 'top',
  },
  error: {
    color: 'red',
    marginBottom: 8,
  },
  deleteButtonContainer: {
    marginTop: 16,
  },
});
