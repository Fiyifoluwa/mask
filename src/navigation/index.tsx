import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator, TransitionPresets} from '@react-navigation/stack';
import {Home} from '../screens/Home';
import {AddItem} from '../screens/AddItem';
import {EditItem} from '../screens/EditItem';
import {AppStackParamList} from './types';

const Stack = createStackNavigator<AppStackParamList>();

export const modalOptions = {
  presentation: 'modal',
  gestureEnabled: false,
  ...TransitionPresets.ModalSlideFromBottomIOS,
};

const Navigation = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="Home"
          component={Home}
          options={{title: 'Inventory'}}
        />
        <Stack.Screen
          name="AddItem"
          component={AddItem}
          options={{title: 'Add New Item', ...modalOptions}}
        />
        <Stack.Screen
          name="EditItem"
          component={EditItem}
          options={{title: 'Edit Item', ...modalOptions}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Navigation;
