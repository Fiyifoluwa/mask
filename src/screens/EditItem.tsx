import React from 'react';
import {Alert, StyleSheet} from 'react-native';
import {useInventory} from '../context/InventoryContext';
import {InventoryForm} from '../components/InventoryForm';
import {InventoryItem} from '../types';
import {AppStackScreenProps} from '../navigation/types';
import LayoutContainer from '../components/LayoutContainer';
import {lastModifiedAt} from '../utils/functions';
import Box from '../components/Box';
import Text from '../components/Text';
import Toast from 'react-native-toast-message';

type Props = AppStackScreenProps<'EditItem'>;

export const EditItem = ({route, navigation}: Props) => {
  const {
    updateItem,
    deleteItem,
    state: {items},
  } = useInventory();
  const currentItem: InventoryItem = route.params.item;

  const handleSubmit = async (values: Partial<InventoryItem>) => {
    const updatedItem: InventoryItem = {
      ...currentItem,
      ...values,
      lastModified: new Date().toISOString(),
    };

    await updateItem(updatedItem);
    Toast.show({
      type: 'infoToast',
      props: {
        text: 'Item updated successfully',
      },
    });
    navigation.goBack();
  };

  const handleDelete = () => {
    Alert.alert('Delete Item', 'Are you sure you want to delete this item?', [
      {text: 'Cancel', style: 'cancel'},
      {
        text: 'Delete',
        style: 'destructive',
        onPress: async () => {
          await deleteItem(currentItem.id);
          Toast.show({
            type: 'dangerToast',
            props: {
              text: 'Item deleted successfully',
            },
          });
          navigation.goBack();
        },
      },
    ]);
  };

  const lastModified = lastModifiedAt(
    currentItem.lastModified ?? new Date().toISOString(),
  );

  return (
    <LayoutContainer
      backgroundColor="background"
      header
      headerText="Edit Item"
      subHeader="Edit the details of this item">
      <Box style={styles.container}>
        <InventoryForm
          initialValues={currentItem}
          onSubmit={handleSubmit}
          submitButtonText="Update Item"
          existingItems={items}
          currentItemId={currentItem.id}
          handleDelete={handleDelete}
        />
        <Box px="m" alignSelf="center" py="m">
          <Text color="mainText" variant="medium10" textAlign="center">
            Last Modified: {lastModified}
          </Text>
        </Box>
      </Box>
    </LayoutContainer>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 16,
    backgroundColor: '#fff',
  },
  fieldContainer: {
    marginBottom: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 4,
    padding: 8,
  },
  inputError: {
    borderColor: 'red',
  },
  multiline: {
    height: 100,
    textAlignVertical: 'top',
  },
  errorText: {
    color: 'red',
    fontSize: 12,
    marginTop: 4,
  },
  deleteButtonContainer: {
    marginTop: 16,
  },
});
