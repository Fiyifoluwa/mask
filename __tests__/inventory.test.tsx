import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {InventoryProvider} from '../src/context/InventoryContext';
import {Home} from '../src/screens/Home';
import {AddItem} from '../src/screens/AddItem';
import {EditItem} from '../src/screens/EditItem';
import {Alert} from 'react-native';
import {StorageService} from '../src/utils/storage';
import {AppStackScreenProps} from '../src/navigation/types';

type EditItemProps = AppStackScreenProps<'EditItem'>;

const createEditItemMockNav = () => {
  const mockRoute = {
    key: 'EditItem-123',
    name: 'EditItem' as const,
    params: {
      item: {
        id: '1',
        name: 'Test Item',
        totalStock: 10,
        price: 99.99,
        description: 'Test description for item',
      },
    },
  } as EditItemProps['route'];

  const mockNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  } as unknown as EditItemProps['navigation'];

  return {
    mockRoute,
    mockNavigation,
  };
};

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
    goBack: jest.fn(),
  }),
}));

jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn(),
}));

const mockedAsyncStorage = AsyncStorage as jest.Mocked<typeof AsyncStorage>;

const mockItem = {
  id: '1',
  name: 'Test Item',
  totalStock: 10,
  price: 99.99,
  description: 'This is a test item description',
};

describe('Screen Snapshots', () => {
  it('should match Home screen snapshot', () => {
    const {toJSON} = render(
      <NavigationContainer>
        <InventoryProvider>
          <Home />
        </InventoryProvider>
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match AddItem screen snapshot', () => {
    const {toJSON} = render(
      <NavigationContainer>
        <InventoryProvider>
          <AddItem />
        </InventoryProvider>
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match EditItem screen snapshot', () => {
    const mockRoute = createEditItemMockNav();

    const {toJSON} = render(
      <NavigationContainer>
        <InventoryProvider>
          <EditItem
            route={mockRoute.mockRoute}
            navigation={mockRoute.mockNavigation}
          />
        </InventoryProvider>
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Delete Confirmation', () => {
  it('should show confirmation dialog when delete button is pressed', async () => {
    const mockRoute = createEditItemMockNav();

    const mockAlert = jest.spyOn(Alert, 'alert');
    const {getByText} = render(
      <NavigationContainer>
        <InventoryProvider>
          <EditItem
            route={mockRoute.mockRoute}
            navigation={mockRoute.mockNavigation}
          />
        </InventoryProvider>
      </NavigationContainer>,
    );

    const deleteButton = getByText('Delete Item');
    fireEvent.press(deleteButton);

    expect(mockAlert).toHaveBeenCalledWith(
      'Delete Item',
      'Are you sure you want to delete this item?',
      expect.any(Array),
    );
  });
});

describe('Navigation Tests', () => {
  it('should navigate to EditItem screen when pressing an item in the list', () => {
    const navigation = {navigate: jest.fn()};
    const {getByText} = render(
      <NavigationContainer>
        <InventoryProvider>
          <Home />
        </InventoryProvider>
      </NavigationContainer>,
    );

    const itemElement = getByText(mockItem.name);
    fireEvent.press(itemElement);

    expect(navigation.navigate).toHaveBeenCalledWith('EditItem', {
      item: mockItem,
    });
  });
});

describe('AsyncStorage CRUD Operations', () => {
  beforeEach(() => {
    mockedAsyncStorage.getItem.mockClear();
    mockedAsyncStorage.setItem.mockClear();
    mockedAsyncStorage.removeItem.mockClear();
  });

  it('should create an item in storage', async () => {
    await StorageService.addItem(mockItem);
    expect(mockedAsyncStorage.setItem).toHaveBeenCalledWith(
      'inventory',
      expect.any(String),
    );
  });

  it('should retrieve items from storage', async () => {
    mockedAsyncStorage.getItem.mockResolvedValue(JSON.stringify([mockItem]));
    const items = await StorageService.getItems();
    expect(mockedAsyncStorage.getItem).toHaveBeenCalledWith('inventory');
    expect(items).toEqual([mockItem]);
  });

  it('should update an item in storage', async () => {
    const updatedItem = {...mockItem, name: 'Updated Test Item'};
    mockedAsyncStorage.getItem.mockResolvedValue(JSON.stringify([mockItem]));
    await StorageService.updateItem(updatedItem);
    expect(mockedAsyncStorage.setItem).toHaveBeenCalledWith(
      'inventory',
      expect.stringContaining('Updated Test Item'),
    );
  });

  it('should delete an item from storage', async () => {
    mockedAsyncStorage.getItem.mockResolvedValue(JSON.stringify([mockItem]));
    await StorageService.deleteItem(mockItem.id);
    expect(mockedAsyncStorage.setItem).toHaveBeenCalledWith(
      'inventory',
      expect.stringMatching(/^\[\]$/),
    );
  });
});

// Test form validation
describe('Form Validation', () => {
  it('should validate required fields in AddItem form', async () => {
    const {getByText} = render(
      <NavigationContainer>
        <InventoryProvider>
          <AddItem />
        </InventoryProvider>
      </NavigationContainer>,
    );

    const submitButton = getByText('Add Item');
    fireEvent.press(submitButton);

    await waitFor(() => {
      expect(getByText('Name is required')).toBeTruthy();
      expect(getByText('Total stock is required')).toBeTruthy();
      expect(getByText('Price is required')).toBeTruthy();
      expect(
        getByText('Description must have at least three words'),
      ).toBeTruthy();
    });
  });

  it('should validate unique name constraint', async () => {
    const existingItems = [mockItem];
    const {getByPlaceholderText, getByText} = render(
      <NavigationContainer>
        <InventoryProvider>
          <AddItem />
        </InventoryProvider>
      </NavigationContainer>,
    );

    const nameInput = getByPlaceholderText('Item Name');
    fireEvent.changeText(nameInput, mockItem.name);

    await waitFor(() => {
      expect(getByText('Item name must be unique')).toBeTruthy();
    });
  });
});
