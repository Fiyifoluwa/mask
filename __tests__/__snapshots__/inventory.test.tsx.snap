// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Snapshots should match AddItem screen snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Item Name"
        style={
          [
            {
              "borderColor": "#ccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 8,
            },
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        keyboardType="numeric"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Total Stock"
        style={
          [
            {
              "borderColor": "#ccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 8,
            },
            undefined,
          ]
        }
        value="0"
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        keyboardType="decimal-pad"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Price"
        style={
          [
            {
              "borderColor": "#ccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 8,
            },
            undefined,
          ]
        }
        value="0"
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <TextInput
        multiline={true}
        numberOfLines={4}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Description (minimum 3 words)"
        style={
          [
            {
              "borderColor": "#ccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 8,
            },
            {
              "height": 100,
              "textAlignVertical": "top",
            },
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Item
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Screen Snapshots should match EditItem screen snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingVertical": 16,
    }
  }
>
  <RCTScrollView
    style={
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View>
      <View
        style={
          {
            "marginBottom": 16,
          }
        }
      >
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Item Name"
          style={
            [
              {
                "borderColor": "#ccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "padding": 8,
              },
              undefined,
            ]
          }
          value="Test Item"
        />
      </View>
      <View
        style={
          {
            "marginBottom": 16,
          }
        }
      >
        <TextInput
          keyboardType="numeric"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Total Stock"
          style={
            [
              {
                "borderColor": "#ccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "padding": 8,
              },
              undefined,
            ]
          }
          value="10"
        />
      </View>
      <View
        style={
          {
            "marginBottom": 16,
          }
        }
      >
        <TextInput
          keyboardType="decimal-pad"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Price"
          style={
            [
              {
                "borderColor": "#ccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "padding": 8,
              },
              undefined,
            ]
          }
          value="99.99"
        />
      </View>
      <View
        style={
          {
            "marginBottom": 16,
          }
        }
      >
        <TextInput
          multiline={true}
          numberOfLines={4}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Description (minimum 3 words)"
          style={
            [
              {
                "borderColor": "#ccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "padding": 8,
              },
              {
                "height": 100,
                "textAlignVertical": "top",
              },
              undefined,
            ]
          }
          value="Test description for item"
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            disabled={false}
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Update Item
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "marginTop": 16,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "red",
              },
            ]
          }
        >
          Delete Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Screen Snapshots should match Home screen snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        No items in inventory
      </Text>
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <Text
        onLayout={[Function]}
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        No items in inventory
      </Text>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#007AFF",
        "borderRadius": 30,
        "bottom": 20,
        "elevation": 4,
        "opacity": 1,
        "padding": 16,
        "position": "absolute",
        "right": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontWeight": "bold",
        }
      }
    >
      Add Item
    </Text>
  </View>
</View>
`;
